discordbot/OverworldDiscordBot.py

```diff
@@ -0,0 +1,282 @@
+# Needs a discord bot account and a server with all the mark ids as emojis.
+
+import asyncio
+import io
+import json
+import socket
+import sys
+from functools import lru_cache
+from threading import Thread
+from time import sleep
+from enum import Enum
+from distutils.util import strtobool
+
+import discord
+import pokebase as pb
+from discord.ext import commands
+from discord.utils import get
+from PIL import Image
+
+sys.path.append('../')
+from lookups import Util
+from nxreader import SWSHReader
+# from . import Channels
+
+class Channels(Enum):
+    NoChannel = 0
+    EventNotificationChannelIdForShiny = 1
+    EventNotificationChannelIdForBrilliant = 2
+    EventNotificationChannelIdForMark = 3
+    EventNotificationChannelIdForRareMark = 4
+    NotificationChannelForInfo = 5
+    TextNotificationChannelIdForShiny = 6
+    TextNotificationChannelIdForBrilliant = 7
+    TextNotificationChannelIdForMark = 8
+    TextNotificationChannelIdForRareMark = 9
+
+class OverworldDiscordBot(commands.Bot):
+    def __init__(self):
+        # config to refactor out later
+        # prefix for bot commands
+        bot_prefix = "$"
+        # set up commands.Bot with specified prefix
+        super().__init__(command_prefix=bot_prefix)
+
+        # default values for variables as to not throw errors
+        self.thread = None
+        self.thread_running = True
+        self.reader = None
+        self.config = None
+        self.is_configured = False
+
+        # function to run when the bot is "ready"
+        @self.event
+        async def on_ready():
+            # tell user in console that the bot is ready
+            _message = "Overworld Discord Bot has started."
+            await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+            # change bot presence to "Watching some ram for shinies"
+            await self.change_presence(activity=discord.Activity(type=discord.ActivityType.watching, name="some ram for shinies"))
+
```

Comment:
Note for future: save stats and have a command to display them (time running, overworld mons read, shinies. how many of each mark(?), etc.

Reply:
Added as a future improvement in the readme.